### ISSUE TITLE
Asking about "import Comparative_models.CE as CE"

### ISSUE THREAD

**Issue by yobrepus** [Julian-Wyatt/AnoDDPM#3]
In the file "detection.py," I found "import Comparative_models.CE as CE" on line 466. How I can get this package? Moreover, I want to run your model based on the dataset  "MVTec". Would you please advise me on how I can reproduce the result? What the configuration file should look like and the steps to reproduce the result.

**Comment by Julian-Wyatt** [comment]
That first module is the context encoder used to generate comparative statistics for the paper. I wasn't planning on uploading that but I can do if you'd still like it?
So I ran the model on the leather subset of MVTec so you should be able to use the MTec Dataset class and load in the model that way - although when instantiating and using the model throughout I've referred to it as "leather", but you should be able to edit those sections and run it ok - the config file I used looks like (changing the noise_fn for simplex for that model): 
```
{
  "img_size": [256,256],
  "Batch_Size": 1,
  "EPOCHS": 3000,
  "T": 1000,
  "base_channels": 128,
  "beta_schedule": "linear",
  "channel_mults": "",
  "loss-type": "l2",
  "loss_weight": "none",
  "train_start": true,
  "lr": 1e-4,
  "random_slice": true,
  "sample_distance": 600,
  "weight_decay": 0.0,
  "save_imgs":false,
  "save_vids":true,
  "dropout":0,
  "attention_resolutions":"16,8",
  "num_heads":2,
  "num_head_channels":-1,
  "noise_fn":"gauss",
  "dataset":"leather"
}

```


Let me know if this helps and get back in touch if not

**Comment by yobrepus** [comment]
Thank you very much for your advice. Now I can train the model. By the way, Can you explain in detail how to evaluate the model ? I don't know how to use detection.py

**Comment by Julian-Wyatt** [comment]
Firstly, you must ensure the function you want to run is selected at the bottom of the file and linked under the arg file provided.
Since you're running the model on MVTec, the anomalous_metric_calculation function will loop over the leather dataset and record various relevant metrics to a file. This is the primary function used for evaluation. 

Alternatively, if you want to reproduce the graphs for example, then the roc_data and graph_data functions sort this however they would need to be adapted for your dataset and remove redundant comparative models. 

Let me know whether this helps

**Comment by Julian-Wyatt** [comment]
Closing due to no responses

**Comment by Lee-Gao** [comment]
Hello!
Recently，I'm learning the code. I see "import Comparative_models.CE as CE" in some modules. And I can't find the module in the AnoDDPM repo. So I want to know how to get this module?
Looking forward to your reply! Thank you!

**Comment by mingqing666** [comment]
> Thank you very much for your advice. Now I can train the model. By the way, Can you explain in detail how to evaluate the model ? I don't know how to use detection.py

Hello, I am also facing the issue with "import Comparitive_models.CE as CE". I checked this thread but couldn't find the uploaded Comparitive_models. Have you found a solution to it?

**Comment by lwtgithublwt** [comment]
> In the file "detection.py," I found "import Comparative_models.CE as CE" on line 466. How I can get this package? Moreover, I want to run your model based on the dataset "MVTec". Would you please advise me on how I can reproduce the result? What the configuration file should look like and the steps to reproduce the result.

Import Comparative_ Models CE as CE, how did you solve it

**Comment by Julian-Wyatt** [comment]
> > Thank you very much for your advice. Now I can train the model. By the way, Can you explain in detail how to evaluate the model ? I don't know how to use detection.py
> 
> Hello, I am also facing the issue with "import Comparitive_models.CE as CE". I checked this thread but couldn't find the uploaded Comparitive_models. Have you found a solution to it?

I don't believe I have access to that file anymore, just delete the references to it as it is just required for comparing models.
